:::NVLOGv0.1.0 ncf 1594265254.556638241 (ncf.py:171) cpu_info: {"num": 16, "name": "Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz"}
:::NVLOGv0.1.0 ncf 1594265254.562041044 (ncf.py:171) mem_info: {"ram": "62G"}
:::NVLOGv0.1.0 ncf 1594265254.650651693 (ncf.py:171) gpu_info: {"driver_version": "440.100", "num": 1, "name": ["GeForce RTX 2080 Super with Max-Q Design"], "mem": ["7982 MiB"]}
:::NVLOGv0.1.0 ncf 1594265254.652277470 (ncf.py:174) args: {"data": "/data/ncf/cache/ml-20m", "epochs": 2, "batch_size": 240000, "valid_batch_size": 1048576, "factors": 64, "layers": [256, 256, 128, 64], "negative_samples": 4, "learning_rate": 0.0045, "topk": 10, "seed": 1, "threshold": 1.0, "beta1": 0.25, "beta2": 0.5, "eps": 1e-08, "dropout": 0.5, "checkpoint_dir": "/data/checkpoints/", "load_checkpoint_path": null, "mode": "train", "grads_accumulated": 1, "opt_level": "O2", "local_rank": 0, "distributed": false, "world_size": 1}
Saving results to /data/checkpoints/
:::NVLOGv0.1.0 ncf 1594265254.652906179 (ncf.py:184) preproc_hp_sample_eval_replacement: true
:::NVLOGv0.1.0 ncf 1594265254.653180599 (ncf.py:185) input_hp_sample_train_replacement: true
:::NVLOGv0.1.0 ncf 1594265254.653437376 (ncf.py:186) input_step_eval_neg_gen
:::NVLOGv0.1.0 ncf 1594265255.977905989 (ncf.py:194) run_start
:::NVLOGv0.1.0 ncf 1594265256.212034702 (ncf.py:201) preproc_hp_num_eval: 100
:::NVLOGv0.1.0 ncf 1594265256.953748941 (ncf.py:207) input_size: 19861770
:::NVLOGv0.1.0 ncf 1594265257.012696505 (ncf.py:216) input_batch_size: 240000
:::NVLOGv0.1.0 ncf 1594265257.013003111 (ncf.py:217) input_order
:::NVLOGv0.1.0 ncf 1594265257.013456583 (/workspace/examples/ncf/neumf.py:54) model_hp_mf_dim: 64
:::NVLOGv0.1.0 ncf 1594265257.308900595 (/workspace/examples/ncf/neumf.py:62) model_hp_mlp_layer_sizes: [256, 256, 128, 64]
Selected optimization level O2:  FP16 training with FP32 batchnorm and FP32 master weights.

Defaults for this optimization level are:
enabled                : True
opt_level              : O2
cast_model_type        : torch.float16
patch_torch_functions  : False
keep_batchnorm_fp32    : True
master_weights         : True
loss_scale             : dynamic
Processing user overrides (additional kwargs that are not None)...
After processing overrides, optimization options are:
enabled                : True
opt_level              : O2
cast_model_type        : torch.float16
patch_torch_functions  : False
keep_batchnorm_fp32    : False
master_weights         : True
loss_scale             : dynamic
NeuMF(
  (mf_user_embed): Embedding(138493, 64)
  (mf_item_embed): Embedding(26744, 64)
  (mlp_user_embed): Embedding(138493, 128)
  (mlp_item_embed): Embedding(26744, 128)
  (mlp): ModuleList(
    (0): Linear(in_features=256, out_features=256, bias=True)
    (1): Linear(in_features=256, out_features=128, bias=True)
    (2): Linear(in_features=128, out_features=64, bias=True)
  )
  (final): Linear(in_features=128, out_features=1, bias=True)
)
31832577 parameters
:::NVLOGv0.1.0 ncf 1594265257.620241404 (ncf.py:246) opt_learning_rate: 0.0045
:::NVLOGv0.1.0 ncf 1594265257.620612383 (ncf.py:247) opt_name: "Adam"
:::NVLOGv0.1.0 ncf 1594265257.620936871 (ncf.py:248) opt_hp_Adam_beta1: 0.25
:::NVLOGv0.1.0 ncf 1594265257.621255398 (ncf.py:249) opt_hp_Adam_beta2: 0.5
:::NVLOGv0.1.0 ncf 1594265257.621574879 (ncf.py:250) opt_hp_Adam_epsilon: 1e-08
:::NVLOGv0.1.0 ncf 1594265257.621884108 (ncf.py:251) model_hp_loss_fn: "binary_cross_entropy"
:::NVLOGv0.1.0 ncf 1594265257.622190952 (ncf.py:279) train_loop
:::NVLOGv0.1.0 ncf 1594265257.622506142 (ncf.py:282) train_epoch_start: 0
:::NVLOGv0.1.0 ncf 1594265257.622819185 (ncf.py:283) input_hp_num_neg: 4
:::NVLOGv0.1.0 ncf 1594265257.623132944 (ncf.py:284) input_step_train_neg_gen
:::NVLOGv0.1.0 ncf 1594265272.206032038 (ncf.py:318) train_throughput: 6810149.228927881
:::NVLOGv0.1.0 ncf 1594265272.206362009 (ncf.py:319) train_epoch_stop: 0
:::NVLOGv0.1.0 ncf 1594265272.206648350 (ncf.py:320) eval_start: 0
../aten/src/ATen/native/IndexingUtils.h:20: UserWarning: indexing with dtype torch.uint8 is now deprecated, please use a dtype torch.bool instead.
:::NVLOGv0.1.0 ncf 1594265272.800926685 (ncf.py:155) eval_size: {"epoch": 0, "value": 13987793}
:::NVLOGv0.1.0 ncf 1594265272.801266670 (ncf.py:156) eval_hp_num_users: 138493
:::NVLOGv0.1.0 ncf 1594265272.801567793 (ncf.py:157) eval_hp_num_neg: 100
Epoch 0: HR@10 = 0.8981, NDCG@10 = 0.6483, train_time = 14.58, val_time = 0.60
:::NVLOGv0.1.0 ncf 1594265272.801994324 (ncf.py:333) eval_accuracy: {"epoch": 0, "value": 0.8981320355541436}
:::NVLOGv0.1.0 ncf 1594265272.802283049 (ncf.py:334) eval_target: 1.0
:::NVLOGv0.1.0 ncf 1594265272.802551508 (ncf.py:335) eval_stop: 0
:::NVLOGv0.1.0 ncf 1594265272.802825212 (ncf.py:340) eval_throughput: 23466552.37764038
New best hr! Saving the model to:  /data/checkpoints/model.pth
:::NVLOGv0.1.0 ncf 1594265272.942746401 (ncf.py:282) train_epoch_start: 1
:::NVLOGv0.1.0 ncf 1594265272.943110466 (ncf.py:283) input_hp_num_neg: 4
:::NVLOGv0.1.0 ncf 1594265272.943398952 (ncf.py:284) input_step_train_neg_gen
Traceback (most recent call last):
  File "ncf.py", line 366, in <module>
    main()
  File "ncf.py", line 288, in main
    epoch_users, epoch_items, epoch_label = dataloading.prepare_epoch_train_data(train_ratings, nb_items, args)
  File "/workspace/examples/ncf/dataloading.py", line 100, in prepare_epoch_train_data
    epoch_indices = torch.randperm(train_users_end - train_users_begin, device='cuda:{}'.format(args.local_rank))
RuntimeError: CUDA out of memory. Tried to allocate 1.48 GiB (GPU 0; 7.80 GiB total capacity; 6.33 GiB already allocated; 172.44 MiB free; 6.80 GiB reserved in total by PyTorch)
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/conda/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/conda/lib/python3.6/site-packages/torch/distributed/launch.py", line 253, in <module>
    main()
  File "/opt/conda/lib/python3.6/site-packages/torch/distributed/launch.py", line 249, in main
    cmd=cmd)
subprocess.CalledProcessError: Command '['/opt/conda/bin/python', '-u', 'ncf.py', '--local_rank=0', '--data', '/data/ncf/cache/ml-20m', '--epochs', '2', '--batch_size', '240000', '--opt_level', 'O2']' returned non-zero exit status 1.
DONE!
